deploy-service/common/src/main/java/com/pinterest/deployservice/rodimus/RodimusManagerImpl.java

```diff
@@ -68,31 +73,97 @@ public boolean shouldSkipClass(Class<?> c) {
         }
     }
 
+    private boolean refreshCachedKey() throws Exception {
+        String prevKnoxKey = this.catchedKey;
+
+        if (this.fsKnox != null) {
+            this.catchedKey = new String(this.fsKnox.getPrimaryKey());
+        }
+
+        if ( prevKnoxKey == null )
+        {
+            return this.catchedKey != null;
+        }else{
+            return ! prevKnoxKey.equals( this.catchedKey );
+        }
+        
+    }
+
+    private void setAuthorization() throws Exception {
+        if (this.fsKnox != null) {
```

Comment:
fxKnox check is not needed here.

Reply:
left it there in the event we want to add another method to get the token. please advice if you prefer it removed.